docs/getting-started.md

```diff
@@ -86,6 +97,78 @@ Built into the framework of the UI is a side navigation bar, along with the top,
+
+### Client-Driven Data
+
+![Client-Driven Data](./assets/images/design-pattern-client.png){data-zoomable}
+_Example workflow to demonstrate the "Client-Driven Data" design pattern._
+
+In Dashboard 2.0 we can configure a given node type to ["Accept Client Data"](/user/sidebar.html#client-data) from the sidebar:
+
+<img data-zoomable style="max-width: 400px; margin: auto;" src="/images/dashboard-sidebar-clientdata.png" alt="Screenshot of an example 'Client Data' tab"/>
+<em>Screenshot of an example "Client Data" tab</em>
+
+If "Include Client Data" is toggled on, then _all_ `msg` objects emitted from _all_ nodes will contain a `msg._client` object, which will at a minimum detail the `socketId` for the connected client. It is possible to add on more data to this object, such as a username, email address, or other unique identifier with Dashboard plugins, e.g. the [FlowFuse User Plugin](https://flowfuse.com/blog/2024/04/displaying-logged-in-users-on-dashboard/).
+
+The "Accept Client Data" table allow configuration over which node types will pay attention to any provided `msg._client` information. Any `msg` sent _to_ one of these nodes can include a `msg._client` value to specify a particular connection (e.g. username, socket ID) that the data should be sent to, rather than to all clients.
+
+For users familiar with the original Node-RED Dashboard, you'll recognise this pattern from what you could do with `ui-notification` and `ui-control`, now, in Dashboard 2.0, it's possible for _all_ widgets.
+
+The key here is that data is generally injected into a node as a consequence of a user action, e.g. clicking a button, viewing a page, or submitting a form, and the responding data is sent _only_ back to that user.
+
+An easy example of this design pattern in Dashboard 2.0 is to utilise the [UI Event](./nodes/widgets/ui-event.md) node. The `ui-event` node emits a `msg` when a user loads a page. Within the `msg` is a full `msg._client` data object available for that client's connection. If this message is then sent onto another node that accept client data, then that full `msg` will _only_ be sent to that specified client.
+
+### Example
+
+Here we have a flow that will produce some client-defined data, and some shared data. 
+
+<video controls>
+    <source src="./assets/videos/demo-design-patterns.mp4" type="video/mp4">
+    Your browser does not support the video tag.
+</video>
+
+In the video above we see that in some cases, data is sent to just the client that triggered it (e.g. button clicks), and in others, the data is shared across all client sessions (e.g. the visualisation of slider value on the chart).
+
+If you'd like to play with this example, the flow is as follows:
+
+<FlowViewer :flow="examples['design-patterns']" height="425px" style="margin-bottom: 24px;"/>
```

Comment:
In testing this, the Click Me button and "Your Latest Button Click:" ui-text was updating on BOTH.  I did not have `ui-text` or `ui-button` [x] checked in the side-bar client data (it worked once i set these).  
Should we be making a reference to that here?

Reply:
Ah, I think this is probably lost when we do the duplicate `ui-base` check - may need to improve that merger to account for the fact this value will probably differ `ui-base` to `ui-base`, rather than just keeping the _older_ `ui-base` as we currently do.

Reply:
Issue Opened: https://github.com/FlowFuse/node-red-dashboard/issues/884